store/src/java/com/zimbra/cs/store/StoreManager.java

```diff
@@ -68,18 +92,101 @@ public static StoreManager getInstance(String className) {
         return sInstance;
     }
 
-
+    /**
+     *  Reset StoreManager for current volume
+     *  TODO: Add documentation here
+     * @throws ServiceException
+     * @throws IOException
+     */
+    public static void resetStoreManager() throws ServiceException, IOException {
+        if (Provisioning.getInstance().getLocalServer().isSMRuntimeSwitchEnabled()) {
+            ZimbraLog.store.error("isSMRunTimeSwitchEnabled is not enabled");
+        }
+        if(ImapDaemon.isRunningImapInsideMailboxd()) {
+            synchronized (StoreManager.class) {
+                ZimbraLog.store.error("resetting StoreManager: Thread: %s", Thread.currentThread().getName());
+                if (sInstance != null) {
+                    ZimbraLog.store.debug("currentStoreManager:%s", sInstance.getClass().getName());
+                }
+                sInstance = null;
+                StoreManager storeManager = getInstance(LC.zimbra_class_store.value());
+                ZimbraLog.store.error("resetting StoreManager: Thread: %s, SM: %s", Thread.currentThread().getName(), LC.zimbra_class_store.value());
+                storeManager.startup();
+                return;
+            }
+        }
+        throw ServiceException.OPERATION_DENIED("Not support when Running Imap Inside Mailboxd");
```

Comment:
message does not sound right. May be `Not supported when running IMAP inside mailbioxd`

Reply:
Done